v2/slogger/appender.go

```diff
@@ -25,26 +25,68 @@ type Appender interface {
 	Flush() error
 }
 
-func FormatLog(log *Log) string {
-	year, month, day := log.Timestamp.Date()
-	hour, min, sec := log.Timestamp.Clock()
-	millisec := log.Timestamp.Nanosecond() / 1000000
+var FormatLogFunc = FormatLog
+
+func SetFormatLogFunc(f func(log *Log) string) {
+	FormatLogFunc = f
```

Comment:
not thread-safe

Reply:
Agreed - however I was following an existing pattern - see `IgnoreThisFilenameToo`.
Also, I think it's unlikely that this will be called outside of the initial consumer setting.
Let me know whether we should either:
1 - add a comment that this isn't thread-safe
2 - go over the existing code to spot unsafe methods

Reply:
not #1. 

Please make this thread-safe.  With the number of people working on the agent code base increasing we need to be more defensive against races.

and #2 is not necessary as those are not regressions.  However, if you have the time, yes, please scan the code.

Reply:
fixed!

Reply:
Could you file a ticket for anything you've already identified for #2?

Reply:
done!